src/asmcnc/apps/drywall_cutter_app/widget_drywall_shape_display.py

```diff
@@ -176,14 +205,15 @@ def __init__(self, **kwargs):
         self.m = kwargs['machine']
         self.sm = kwargs['screen_manager']
         self.dwt_config = kwargs['dwt_config']
+        self.engine = kwargs['engine']
 
         self.d_input.bind(text=self.d_input_change) # Diameter of circle
         self.l_input.bind(text=self.l_input_change) # Length of line
         self.r_input.bind(text=self.r_input_change) # Radius of corners
         self.x_input.bind(text=self.x_input_change) # Square/rectangle x length
         self.y_input.bind(text=self.y_input_change) # Square/rectangle y length
 
-        Clock.schedule_interval(self.poll_position, 0.1)
+        Clock.schedule_interval(self.check_datum_and_extents, 0.1)
```

Comment:
Should this be on a 0.1s clock? If you can make this event based, it would probably be better. There's a lot of logic going on, and this call with Clock is on the kivy UI thread. Is this to keep the bumpers accurate if the machine is being moved? The on_release of the move widgets buttons and when values are changed could be the executor.

Reply:
I created a ticket for that